packages/evals/src/attachListeners.ts

```diff
@@ -1,18 +1,32 @@
+import type { Mastra } from '@mastra/core';
 import { AvailableHooks, registerHook } from '@mastra/core/hooks';
+import { MastraStorage } from '@mastra/core/storage';
 import { mkdirSync, appendFile } from 'fs';
 import { join } from 'path';
 
 import { GLOBAL_RUN_ID_ENV_KEY } from './constants';
 
-export async function attachListeners() {
+export async function attachListeners(mastra: Mastra) {
   const dotMastraPath = join(process.cwd(), '.mastra');
 
   try {
     mkdirSync(dotMastraPath);
   } catch (error) {}
 
-  registerHook(AvailableHooks.ON_EVALUATION, traceObject => {
+  registerHook(AvailableHooks.ON_EVALUATION, async traceObject => {
     appendFile(join(dotMastraPath, 'evals.json'), JSON.stringify(traceObject) + '\n', () => {});
+    if (mastra?.memory?.storage) {
+      await mastra.memory.storage.insert({
+        tableName: MastraStorage.TABLE_EVALS,
+        record: {
+          result: JSON.stringify(traceObject.result),
+          meta: JSON.stringify(traceObject.meta),
+          input: traceObject.input,
+          output: traceObject.output,
+          createdAt: new Date().toISOString(),
+        },
+      });
+    }
```

Comment:
Do we still need an ON_EVALUATION hook? Both dev and testing will call the evaluate function, so I'm thinking we could just insert the evals in that function without going to this hook first.

Reply:
For CI/CD use case, not really but we need this hook when doing realtime evals to keep things decoupled. We can store the necessary info to run the eval and run it whenever we can